src/modelfree/policies/wrappers.py

```diff
@@ -36,3 +36,144 @@ def predict(self, observation, state=None, mask=None, deterministic=False):
+                # and thus had its state value set to None. Stateful policies can take in
+                # None as a state value, but it can't accommodate some of its envs having
+                # arrays and others being None
+                env_mask[i] = False
+
+    # Fill in `filler_value` for all indices of state_arr where env_mask is False
+    filled_state = _array_mask_assign(state_arr,
+                                      [not el for el in env_mask],
+                                      filler_value)
+    return filled_state
+
+
+def _array_mask_assign(arr, mask, vals):
+    """A helper method for basically doing Numpy-style mask assignment on a Python array.
+    The `mask` variable contains boolean True values at all locations within `vals` that we
+    want to copy over to `arr`. If `vals` is not the same first-dimension as mask, it will
+    be broadcast to all locations that `mask` specifies.
+    :return:
+    """
+    arr_copy = arr.copy()
+    # Check the first-dimension length of vals, checking for numpy array, python arr, and None
+    if vals is None:
+        length = None
+    else:
+        try:
+            length = vals.shape[0]
+        except AttributeError:
+            length = len(vals)
+    # If the first dimension is not the same as the mask, assume we want this value
+    # tiled for every True-valued location in env_mask
+    tile_val = length != len(mask)
+    for i in range(len(arr_copy)):
+        if mask[i]:
+            if tile_val:
+                arr_copy[i] = vals
+            else:
+                arr_copy[i] = vals[i]
+    return arr_copy
+
+
+def waterfall_get(obj, possible_attributes):
+    ret_val = None
+    for attribute in possible_attributes:
+        ret_val = obj.__dict__.get(attribute, None)
+        if ret_val is not None:
+            break
+    return ret_val
+
+
+class MultiPolicyWrapper(DummyModel):
+    def __init__(self, policies, num_envs, debug=False):
+        super().__init__(policies[0], policies[0].sess)
+        self.policies = policies
+        # num_envs is kept as a parameter because you need it to construct
+        # self.current_env_policies which makes sense to do the first time at initialization
+        self.num_envs = num_envs
+        self.debug = debug
+        reference_action_space = waterfall_get(self.policies[0].policy,  ['ac_space',
+                                                                          'action_space'])
+        reference_observation_space = waterfall_get(self.policies[0].policy, ['ob_space',
+                                                                              'observation_space'])
+        self.action_space_shape = reference_action_space.shape
+        self.obs_space_shape = reference_observation_space.shape
+        for p in self.policies:
+            err_txt = "All policies must have the same {} space"
+            action_space = waterfall_get(p.policy, ['ac_space', 'action_space']).shape
```

Comment:
The goal of this was to do a minimal sanity check rather than a completely exhaustive one, but, sure, you could add a bunch of additional checks here as well